modules/markdown/markdown.go

```diff
@@ -130,46 +193,50 @@ func (r *Renderer) AutoLink(out *bytes.Buffer, link []byte, kind int) {
 		r.Renderer.AutoLink(out, link, kind)
 		return
 	}
+	r.Renderer.AutoLink(out, link, kind)
+}
 
-	// Since this method could only possibly serve one link at a time,
-	// we do not need to find all.
-	if bytes.HasPrefix(link, []byte(setting.AppUrl)) {
-		m := CommitPattern.Find(link)
-		if m != nil {
-			m = bytes.TrimSpace(m)
-			i := strings.Index(string(m), "commit/")
-			j := strings.Index(string(m), "#")
-			if j == -1 {
-				j = len(m)
-			}
-			out.WriteString(fmt.Sprintf(` <code><a href="%s">%s</a></code>`, m, base.ShortSha(string(m[i+7:j]))))
-			return
-		}
-
-		m = IssueFullPattern.Find(link)
-		if m != nil {
-			m = bytes.TrimSpace(m)
-			i := strings.Index(string(m), "issues/")
-			j := strings.Index(string(m), "#")
-			if j == -1 {
-				j = len(m)
-			}
-			out.WriteString(fmt.Sprintf(`<a href="%s">#%s</a>`, m, base.ShortSha(string(m[i+7:j]))))
-			return
-		}
+func (options *Renderer) List(out *bytes.Buffer, text func() bool, flags int) {
+	marker := out.Len()
+	if out.Len() > 0 {
+		out.WriteByte('\n')
 	}
 
-	r.Renderer.AutoLink(out, link, kind)
+	if flags&blackfriday.LIST_TYPE_DEFINITION != 0 {
+		out.WriteString("<dl>")
+	} else if flags&blackfriday.LIST_TYPE_ORDERED != 0 {
+		out.WriteString("<ol class='ui list'>")
+	} else {
+		out.WriteString("<ul class='ui list'>")
+	}
+	if !text() {
+		out.Truncate(marker)
+		return
+	}
+	if flags&blackfriday.LIST_TYPE_DEFINITION != 0 {
+		out.WriteString("</dl>\n")
+	} else if flags&blackfriday.LIST_TYPE_ORDERED != 0 {
+		out.WriteString("</ol>\n")
+	} else {
+		out.WriteString("</ul>\n")
+	}
 }
 
 // ListItem defines how list items should be processed to produce corresponding HTML elements.
 func (options *Renderer) ListItem(out *bytes.Buffer, text []byte, flags int) {
 	// Detect procedures to draw checkboxes.
+	prefix := ""
+	if bytes.HasPrefix(text, []byte("<p>")) {
+		prefix = "<p>"
+	}
 	switch {
-	case bytes.HasPrefix(text, []byte("[ ] ")):
-		text = append([]byte(`<input type="checkbox" disabled="" />`), text[3:]...)
-	case bytes.HasPrefix(text, []byte("[x] ")):
-		text = append([]byte(`<input type="checkbox" disabled="" checked="" />`), text[3:]...)
+	case bytes.HasPrefix(text, []byte(prefix+"[ ] ")):
+		text = append([]byte(`<div class="ui fitted read-only checkbox"><input type="checkbox" /><label /></div>`), text[3+len(prefix):]...)
```

Comment:
It's a bug, so should be fixed in this PR. I will take a look on what caused it when I have a little time. 👍 

For future contributions, I think it's a good idea to split big features in more PRs. It makes it easier and faster to review.

Let me know if you need help with this. We can push your branch to the main repo, so more people would have write access to it.